src/shims/windows/foreign_items.rs

```diff
@@ -509,9 +509,14 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 };
 
                 // FIXME: use non-lossy conversion
-                this.set_thread_name(thread, String::from_utf16_lossy(&name).into_bytes());
+                let res = match this
+                    .set_thread_name(thread, String::from_utf16_lossy(&name).into_bytes())
+                {
+                    Ok(_) => Scalar::from_u32(0),
+                    Err(_) => Scalar::from_u32(3489660936),
```

Comment:
What is this random number?^^

Please use a named constant instead.

Reply:
A magic undocumented number I wrote you in the issue about. If you write in a binary form then you will find that two higher bits are set to `0b11` which indicates an error. The next bytes are for the reversed and customer bits (don't remember the order, but it doesn't matter) are zeroes. The facility is zero too, and the error code is `0x1000`. That's all I can say to you about that magic number.

It's not documented at all, neither in the function returns, nor as a well known error code.

And just for your info, in Windows there are two very similar date types, ` NTSTATUS` in the kernel space and `HRESULT` in the user space. Both are bit fields rather than just numbers, but both follow the same rule as everywhere, a negative number means an error. Positive numbers can mean success or a warning depending on the 30 bit.

Reply:
Where did you get that number from?

Reply:
It's the NTSTATUS [`STATUS_INVALID_HANDLE`](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/596a1078-e883-4972-9bbc-49e60bebca55) (`0xC0000008`) encoded as a [HRESULT](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/0642cb2f-2075-4469-918c-4441e69c548a) by setting the `N` bit. Which gives the value `0xD0000008`.

Reply:
> Where did you get that number from?

I got it by writing an app that used the thread description functions and feeding them invalid thread handles.

Reply:
Please use a `const STATUS_INVALID_HANDLE_HRESULT = 0xD0000008` to make this more readable.